servicetalk-http-api/src/main/java/io/servicetalk/http/api/ContentCodings.java

```diff
@@ -0,0 +1,118 @@
+/*
+ * Copyright © 2020 Apple Inc. and the ServiceTalk project authors
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package io.servicetalk.http.api;
+
+import java.util.Collection;
+import java.util.HashSet;
+import java.util.Set;
+import javax.annotation.Nullable;
+
+import static java.util.Arrays.asList;
+import static java.util.Collections.unmodifiableSet;
+import static java.util.Objects.requireNonNull;
+
+/**
+ * Default available encoding implementations.
+ * Encoding {@link #none()} is always supported regardless of the client or server settings.
+ *
+ * {@link #all()} is a set that includes all default encodings {@link #deflate()} and {@link #gzip()}.
+ */
+public final class ContentCodings {
+
+    static final CharSequence GZIP_HEADER = "gzip";
+    static final CharSequence DEFLATE_HEADER = "deflate";
+
+    private static final ContentCoding NONE =
+            new DefaultContentCoding("identity", new IdentityContentCodec());
+
+    private static final ContentCoding GZIP =
+            new DefaultContentCoding("gzip", new GzipContentCodec());
+
+    private static final ContentCoding DEFLATE =
+            new DefaultContentCoding("deflate", new DeflateContentCodec());
+
+    private static final Set<ContentCoding> ALL =
+            unmodifiableSet(new HashSet<>(asList(NONE, GZIP, DEFLATE)));
+
+    private ContentCodings() {
+    }
+
+    /**
+     * Returns the default, always supported 'identity' {@link ContentCoding}.
+     * @return the default, always supported 'identity' {@link ContentCoding}
+     */
+    public static ContentCoding none() {
+        return NONE;
+    }
+
+    /**
+     * Returns a GZIP based {@link ContentCoding} backed by {@link java.util.zip.Inflater}.
+     * @return a GZIP based {@link ContentCoding} backed by {@link java.util.zip.Inflater}
+     */
+    public static ContentCoding gzip() {
+        return GZIP;
+    }
+
+    /**
+     * Returns a DEFLATE based {@link ContentCoding} backed by {@link java.util.zip.Inflater}.
+     * @return a DEFLATE based {@link ContentCoding} backed by {@link java.util.zip.Inflater}
+     */
+    public static ContentCoding deflate() {
```

Comment:
Nice topic, hmm I didn't TBH, and even though I want to keep this simple, it might worth doing that sooner rather than later to have a complete API at least instead of possibly breaking it later. Good catch, especially the zip bomb.

Reply:
The `ContentCodings` factory class is good for discoverability. We can keep it, but instead of a `ContentCoding` it may return builders for `deflate` and `gzip`. 
Look it the approach in `HttpProtocolConfigs`.